packages/eas-cli/src/credentials/ios/appstore/contractMessages.ts

```diff
@@ -0,0 +1,85 @@
+import { ITCAgreements, RequestContext } from '@expo/apple-utils';
+import chalk from 'chalk';
+import { Ora } from 'ora';
+
+import log from '../../../log';
+import { convertHTMLToASCII } from '../utils/convertHTMLToASCII';
+
+async function getContractStatusAsync(
+  context: RequestContext
+): Promise<ITCAgreements.ITCContractStatus | null> {
+  try {
+    const capabilities = await ITCAgreements.getCapabilitiesAsync(context);
+    return capabilities?.contractStatus ?? null;
+  } catch (error) {
+    log.warn(`Failed to get iTunes contract status: ${error.message}`);
+    return null;
+  }
+}
+
+async function getContractMessagesAsync(context: RequestContext) {
+  try {
+    return await ITCAgreements.getContractMessagesAsync(context);
+  } catch (error) {
+    log.warn(`Failed to get iTunes contract messages: ${error.message}`);
+    return null;
+  }
+}
+
+async function getRequiredContractMessagesAsync(context: RequestContext) {
+  const status = await getContractStatusAsync(context);
+
+  switch (status) {
+    // The user can freely create an app
+    case 'FREE_APP_AGREEMENT_ACTIVE':
+    case 'PAID_APP_AGREEMENT_ACTIVE':
+      return false;
+    // The user cannot create an app until they've resolved contract issues on ASC.
+    case 'FREE_APP_AGREEMENT_OUTDATED':
+    case 'PAID_APP_AGREEMENT_OUTDATED':
+    case 'EXPIRED_MEMBERSHIP':
+      return (await getContractMessagesAsync(context)) ?? [];
+    default:
+      // TODO: Maybe a silent analytic would be better
```

Comment:
yeah this would be useful to report. cc @fson / @animeeples for thoughts on what we should be using to track these sorts of errors